Analisis Kedatangan Wisatawan \
(Ni Kadek Emik Sapitri)

# Data

In [2]:
import numpy as np
import pandas as pd

data_balitourist = pd.read_csv('bali_tourist.csv')
data_balitourist.head()

year           area     country  January  February  March    April    May  \
0  2019  ASIA PACIFIC   AUSTRALIAN    95536     67474  83002  90265.0  98625   
1  2019  ASIA PACIFIC       BHUTAN       60        30     47     38.0     31   
2  2019  ASIA PACIFIC   BANGLADESH     2262      2094   2489   2264.0   1938   
3  2019  ASIA PACIFIC         FIJI       91        73     72     77.0     42   
4  2019  ASIA PACIFIC     HONGKONG     1313      1367    764      6.0      1   

     June    July  August  September  October  November  December  
0  110815  123293  118556     124416   119382    105312    110711  
1      31      46      67         43       32        56        60  
2    4265    2656    2307       2035     2371      2164      2671  
3      62    2656      80         68       78        89       189  
4       5       2       1          0       11         0        45

In [3]:
data_balitourist.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 362 entries, 0 to 361
Data columns (total 15 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   year       362 non-null    int64  
 1   area       362 non-null    object 
 2   country    362 non-null    object 
 3   January    362 non-null    int64  
 4   February   362 non-null    int64  
 5   March      362 non-null    int64  
 6   April      360 non-null    float64
 7   May        362 non-null    int64  
 8   June       362 non-null    int64  
 9   July       362 non-null    int64  
 10  August     362 non-null    int64  
 11  September  362 non-null    int64  
 12  October    362 non-null    int64  
 13  November   362 non-null    int64  
 14  December   362 non-null    int64  
dtypes: float64(1), int64(12), object(2)
memory usage: 42.6+ KB


# Berapa total turis yang datang ke Bali pada tahun 2018 dan 2019?

In [4]:
kolom_bulan = ['January', 'February', 'March', 'April',
               'May', 'June', 'July', 'August',
               'September', 'October', 'November', 'December']

total_perbulan = data_balitourist.groupby('year')[kolom_bulan].sum()
total_perbulan

January  February   March     April     May    June    July  August  \
year                                                                        
2018   358065    452423  492678  516777.0  528512  544550  624366  573766   
2019   456218    437537  449637  476327.0  485634  549751  594279  616706   

      September  October  November  December  
year                                          
2018     555903   517889    407213    501211  
2019     590565   568064    497925    552402

In [5]:
total_turis = total_perbulan.sum(axis=1)
total_turis

year
2018    6073353.0
2019    6275045.0
dtype: float64

Total turis yang datang ke Bali pada tahun 2018 sebanyak 6073353 orang dan pada tahun 2019 sebanyak 6275045 orang.

# Berapa rata-rata turis yang datang ke Bali pada tahun 2018 dan 2019?

In [6]:
mean_turis = total_perbulan.mean(axis=1)
mean_turis

year
2018    506112.750000
2019    522920.416667
dtype: float64

Rata-rata turis yang datang ke Bali pada tahun 2018 adalah sekitar 506113 orang per bulan dan pada tahun 2019 sekitar 522920 orang per bulan.

# Pada bulan apakah di tahun 2018 dan 2019 di mana jumlah total turis paling banyak di antara bulan lainnya?

In [7]:
total_perbulan_max = total_perbulan.idxmax(axis=1)
total_perbulan_max

year
2018      July
2019    August
dtype: object

Jumlah total turis paling banyak di tahun 2018 ada di bulan Juli dan tahun 2019 di bulan Agustus.

# Pada bulan apakah di tahun 2018 dan 2019 di mana jumlah total turis paling sedikit di antara bulan lainnya?

In [8]:
total_perbulan_min = total_perbulan.idxmin(axis=1)
total_perbulan_min

year
2018     January
2019    February
dtype: object

Jumlah total turis paling sedikit di tahun 2018 ada di bulan Januari dan tahun 2019 di bulan Februari.

# Area manakah yang memiliki jumlah turis yang datang ke Bali terbanyak pada tahun 2018 dan 2019?

In [9]:
total_perarea = data_balitourist.groupby(['year', 'area'])[kolom_bulan].sum().sum(axis=1)
total_perarea

year  area               
2018  AFRICA                   54491.0
      AMERICA                 367639.0
      ASEAN                   514324.0
      ASIA PACIFIC           3653078.0
      EUROPE                 1392795.0
      MIDDLE EAST              60577.0
      OTHER NATIONALITIES      30449.0
2019  AFRICA                   66474.0
      AMERICA                 415262.0
      ASEAN                   579627.0
      ASIA PACIFIC           3614482.0
      EUROPE                 1485287.0
      MIDDLE EAST              75659.0
      OTHER NATIONALITIES      38254.0
dtype: float64

In [10]:
area_total = total_perarea.groupby('year').idxmax()
area_total

year
2018    (2018, ASIA PACIFIC )
2019    (2019, ASIA PACIFIC )
dtype: object

Asia Pasifik menjadi area yang memiliki jumlah turis yang datang ke Bali terbanyak pada tahun 2018 dan 2019.

# Rangkuman hasil

In [11]:
rangkuman = pd.concat([total_perbulan, total_perbulan_max, total_perbulan_min, total_turis, mean_turis], axis=1)
rangkuman

January  February   March     April     May    June    July  August  \
year                                                                        
2018   358065    452423  492678  516777.0  528512  544550  624366  573766   
2019   456218    437537  449637  476327.0  485634  549751  594279  616706   

      September  October  November  December       0         1          2  \
year                                                                        
2018     555903   517889    407213    501211    July   January  6073353.0   
2019     590565   568064    497925    552402  August  February  6275045.0   

                  3  
year                 
2018  506112.750000  
2019  522920.416667

In [12]:
#cek nama kolom
print(rangkuman.columns)

Index([  'January',  'February',     'March',     'April',       'May',
            'June',      'July',    'August', 'September',   'October',
        'November',  'December',           0,           1,           2,
                 3],
      dtype='object')


In [13]:
#rapikan

rangkuman_final = rangkuman.rename(columns={0 : 'Max', 1 : 'Min', 2 : 'Total Kedatangan dalam Setahun', 3 : 'Rata-rata Kedatangan dalam Setahun'}) 
rangkuman_final

January  February   March     April     May    June    July  August  \
year                                                                        
2018   358065    452423  492678  516777.0  528512  544550  624366  573766   
2019   456218    437537  449637  476327.0  485634  549751  594279  616706   

      September  October  November  December     Max       Min  \
year                                                             
2018     555903   517889    407213    501211    July   January   
2019     590565   568064    497925    552402  August  February   

      Total Kedatangan dalam Setahun  Rata-rata Kedatangan dalam Setahun  
year                                                                      
2018                       6073353.0                       506112.750000  
2019                       6275045.0                       522920.416667